src/protocols/ab/eip_cip.c

```diff
@@ -627,6 +634,13 @@ int build_read_request_unconnected(ab_tag_p tag, int byte_offset)
     /* allow packing if the tag allows it. */
     req->allow_packing = tag->allow_packing;
 
+    /* set the response size to the size of data from the previous read */
+    req->response_capacity = tag->size;
```

Comment:
Same comment as previous.   The request response size is limited to what you can force in a single response packet (~4k for ControlLogix with newer network modules or a 1756-L8x CPU, ~500 bytes for other Rockwell CIP PLCs, and closer to 256 bytes for PCCC/CSP PLCs like PLC/5).  So I could have a tag with a size of 16k but only get back 4k at a time.

Reply:
I understood that fragmented reads like you say are available on some of the Rockwell PLCs, although I have no experience with them myself. This solution should be able to coexist with the fragmented reads that these PLCs do. It will not be as efficient as using fragmented reads, but for PLCs which dont support this, it will be an improvement over no packing, assuming we have a good way of working out how much data the PLC will return.